docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,685 @@
+- A strictly privacy respecting approach (GDPR)
+
+An approach to tackle the identity and privacy challenge that has been gaining momentum in the recent years is the SSI approach, that, coupled with DLT technology, has been capturing the attention of both the private and public sector. 
+
+The Self-Sovereign Identity (SSI) relays on two building blocks: decentralized identifiers (DIDs) and verifiable credentials (VC). This ADR is about describing the DID implementation in a cosmos-sdk based blockchain.
+
+The goal of this ADR is to define a foundation for the necessary components to realize the Cosmos Cash objectives and on the same time to keep the implementation of the DID fully compliant with the W3C specifications. **Successive iterations will address API ergonomics and standard compatibility issues.** 
+
+## Decision
+
+
+The DID W3C specification are designed using the "open world assumption" approach for data modelling. 
+For the Cosmos Cash implementation we will be following the did-core specification: custom properties will require a fork and a different implementation of the cosmos-cash did module.
+
+The following is an example of a DID Document:
+
+```json
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cosmos:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+    "verificationMethods" : [
+        {
+            "id": "did:cosmos:cosmos:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#key-1",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:cosmos:cosmos:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cosmos1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        }
+    ],
+    "authentication": [
+        "did:cosmos:cosmos:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#key-1"
+    ],
+    "services": [
+        {
+            "id":"agent:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+			"type":"DIDCommMessaging",
+			"serviceEndpoint":"https://agent.xyz/ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+        }
+    ]
+}
+```
+
+### DID Method name
+
+The namestring that shall identify this DID method is: `cosmos`.
+
+A DID that uses this method MUST begin with the following prefix: `did:cosmos`. Per this [DID specification](https://www.w3.org/TR/did-core), this string MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+#### Method Specific Identifier
+
+
+The namespace specific identifier is defined by the following ABNF:
+
+```
+cosmos-did                = "did:cosmos:" cosmos-specific-id-string
+cosmos-specific-id-string = cosmos-chain-name ":" unique-identifier
+cosmos-chain-name         = 1*255id-char
+unique-identifier         = 38*256id-char
+id-char                   = ALPHA / DIGIT / (ALPHA "-") / (DIGIT "-")
+```
+
+For the `unique-identifier` it is RECOMMENDED to use a cosmos account address (without chain prefix and separator) or a UUID 
+
+Examples using a cosmos address: 
```

Comment:
I would put an example from the derived address, so it can better be understood how it looks from the cosmos address to the did Id.